src/components/Dashboard/DashboardItem/ItemBase.js

```diff
@@ -0,0 +1,308 @@
+                        >
+                            <MoreVert />
+                        </IconButton>
+                        <Menu
+                            id={menuID}
+                            anchorEl={menuAnchorEl}
+                            keepMounted
+                            open={Boolean(menuAnchorEl)}
+                            onClose={() => setMenuAnchorEl(null)}
+                        >
+                            {item.menuActions}
+                        </Menu>
+                    </>
+                )}
+            </CardActions>
+        </Card>
+    );
+};
+
+export const DashboardFeedItem = ({ item }) => {
+    const classes = useStyles({ width: item.width, height: item.height });
+    const { t } = useTranslation("dashboard");
+
+    const [origination, date] = item.getOrigination(t);
+    const description = fns.cleanDescription(item.content.description);
+    const user = item.username;
+
+    return (
+        <div
+            className={clsx(
+                item.section === constants.SECTION_NEWS && classes.newsItem,
+                item.section === constants.SECTION_EVENTS && classes.eventsItem
+            )}
+        >
+            <Typography variant="h6" color="primary">
+                <DashboardLink target={item.getLinkTarget()} kind={item.kind}>
+                    {item.content.name}
+                </DashboardLink>
+            </Typography>
+
+            <Typography noWrap color="textSecondary" variant="subtitle2">
+                {`${origination} ${user} on ${date}`}
+            </Typography>
+
+            <Typography variant="body2" component="p">
+                {description || t("noDescriptionProvided")}
+            </Typography>
+        </div>
+    );
+};
+
+export const DashboardVideoItem = ({ item }) => {
+    const classes = useStyles(item);
+
+    return (
+        <iframe
+            className={classes.dashboardVideo}
+            aria-label={item.content.name}
+            title={item.content.name}
+            width="100%"
+            height="100%"
+            src={item.getLinkTarget()}
+            frameBorder="0"
+            controls="1"
+            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture; fullscreen"
+            allowFullScreen
+        ></iframe>
+    );
+};
+
+export const ItemAction = ({ children, tooltipKey }) => {
+    const { t } = useTranslation("dashboard");
+
+    // The nested div prevents props from the Tooltip from getting propagated
+    // down to components that may not support them, like Next.js's Link.
+    return (
+        <Tooltip title={t(tooltipKey)}>
+            <div>{children}</div>
+        </Tooltip>
+    );
+};
+
+export const MenuAction = ({
+    children,
+    ariaLabel,
+    handleClick,
+    tooltipKey,
+}) => {
+    const { t } = useTranslation("dashboard");
+    return (
+        <Tooltip title={t(tooltipKey)}>
+            <MenuItem onClick={handleClick} ariaLabel={ariaLabel}>
```

Comment:
I'm not sure if MUI's `MenuItem`s need aria labels, but if so, this prop should be `aria-label`.